packages/runtime/test/consumption/attributes.test.ts

```diff
@@ -749,6 +749,100 @@ describe(CreateRepositoryAttributeUseCase.name, () => {
         expect(attribute.isDefault).toBeUndefined();
     });
+
+    describe("validation errors for the attribute content", () => {
+        test("should not create a number as GivenName", async () => {
+            const request: CreateRepositoryAttributeRequest = {
+                content: {
+                    value: {
+                        "@type": "GivenName",
+                        value: 5
+                    },
+                    tags: ["tag1", "tag2"]
+                } as any
+            };
+            const result = await services1.consumption.attributes.createRepositoryAttribute(request);
+            expect(result.error.message).toBe("value must be string");
+            expect(result.error.code).toBe("error.runtime.validation.invalidPropertyValue");
+        });
+
+        test("should not create a string as year of BirthDate", async () => {
+            const request: CreateRepositoryAttributeRequest = {
+                content: {
+                    value: {
+                        "@type": "BirthDate",
+                        day: 5,
+                        month: 5,
+                        year: "a-string"
+                    },
+                    tags: ["tag1", "tag2"]
+                } as any
+            };
+            const result = await services1.consumption.attributes.createRepositoryAttribute(request);
+            expect(result.error.message).toBe("year must be number");
+            expect(result.error.code).toBe("error.runtime.validation.invalidPropertyValue");
+        });
+
+        test("should not create a BirthDate with a missing year", async () => {
+            const request: CreateRepositoryAttributeRequest = {
+                content: {
+                    value: {
+                        "@type": "BirthDate",
+                        day: 5,
+                        month: 5
+                    },
+                    tags: ["tag1", "tag2"]
+                } as any
+            };
+            const result = await services1.consumption.attributes.createRepositoryAttribute(request);
+            expect(result.error.message).toBe("must have required property 'year'");
+            expect(result.error.code).toBe("error.runtime.validation.invalidPropertyValue");
+        });
+
+        test("should not create 14 as BirthMonth", async () => {
+            const request: CreateRepositoryAttributeRequest = {
+                content: {
+                    value: {
+                        "@type": "BirthMonth",
+                        value: 14
+                    },
+                    tags: ["tag1", "tag2"]
+                } as any
+            };
+            const result = await services1.consumption.attributes.createRepositoryAttribute(request);
+            expect(result.error.message).toBe("value must be equal to one of the allowed values");
+            expect(result.error.code).toBe("error.runtime.validation.invalidPropertyValue");
+        });
```

Comment:
Does this test causes redundancy as we already have the test `"returns an error when trying to create an invalid BirthDate with violated validation criteria of a single property"` within the file `BirthDate.test.ts`? Within the mentioned test, the message `BirthMonth.value:Number :: must be an integer value between 1 and 12"` is used, which I prefer compared to the more generic `"value must be equal to one of the allowed values"` message. However, I am aware that I am comparing two tests from different libraries (Content and Runtime).

Reply:
The content error message comes from the serval, whereas the error here comes from ajv. I agree that the serval error is better but I'll have to see if there's a good way to improve the error here